plonk-core/src/circuit.rs

```diff
@@ -60,82 +46,108 @@ pub trait GeIntoPubInput<T> {
 /// scalar representation.
 #[derive(CanonicalDeserialize, CanonicalSerialize, derivative::Derivative)]
 #[derivative(Clone, Debug, Default)]
-pub struct PublicInputValue<P>
+pub struct PublicInputValue<F>
 where
-    P: TEModelParameters,
+    F: Field,
 {
     /// Field Values
-    pub(crate) values: Vec<P::BaseField>,
+    pub(crate) values: Vec<F>,
+}
+
+impl<F> From<F> for PublicInputValue<F>
+where
+    F: Field,
+{
+    fn from(p: F) -> PublicInputValue<F> {
+        PublicInputValue { values: vec![p] }
+    }
 }
 
-impl<P> FeIntoPubInput<PublicInputValue<P>> for P::BaseField
+impl<P> GeIntoPubInput<PublicInputValue<P::BaseField>> for TEGroupAffine<P>
 where
     P: TEModelParameters,
 {
     #[inline]
-    fn into_pi(self) -> PublicInputValue<P> {
-        PublicInputValue { values: vec![self] }
+    fn into_pi(self) -> PublicInputValue<P::BaseField> {
+        PublicInputValue {
+            values: vec![self.x, self.y],
+        }
     }
 }
 
-impl<P> GeIntoPubInput<PublicInputValue<P>> for GroupAffine<P>
+impl<P> GeIntoPubInput<PublicInputValue<P::BaseField>> for TEGroupProjective<P>
 where
     P: TEModelParameters,
 {
     #[inline]
-    fn into_pi(self) -> PublicInputValue<P> {
+    fn into_pi(self) -> PublicInputValue<P::BaseField> {
+        GeIntoPubInput::into_pi(self.into_affine())
+    }
+}
+impl<P> GeIntoPubInput<PublicInputValue<P::BaseField>> for SWGroupAffine<P>
+where
+    P: SWModelParameters,
+{
+    #[inline]
+    fn into_pi(self) -> PublicInputValue<P::BaseField> {
         PublicInputValue {
             values: vec![self.x, self.y],
         }
     }
 }
 
-impl<P> GeIntoPubInput<PublicInputValue<P>> for GroupProjective<P>
+impl<P> GeIntoPubInput<PublicInputValue<P::BaseField>> for SWGroupProjective<P>
 where
-    P: TEModelParameters,
+    P: SWModelParameters,
 {
     #[inline]
-    fn into_pi(self) -> PublicInputValue<P> {
+    fn into_pi(self) -> PublicInputValue<P::BaseField> {
         GeIntoPubInput::into_pi(self.into_affine())
     }
 }
 
+/*
+pub enum EmbeddedCurve<F> {
+    TwistedEdwards { a: F, d: F },
+    ShortWeierstrass { a: F, b: F },
+}*/
+
 /// Collection of structs/objects that the Verifier will use in order to
 /// de/serialize data needed for Circuit proof verification.
 /// This structure can be seen as a link between the [`Circuit`] public input
 /// positions and the [`VerifierKey`] that the Verifier needs to use.
 #[derive(CanonicalDeserialize, CanonicalSerialize, derivative::Derivative)]
 #[derivative(
     Clone(bound = ""),
-    Debug(bound = ""),
-    Eq(bound = ""),
-    PartialEq(bound = "")
+    //Debug(bound = ""),
+    //Eq(bound = ""),
+    //PartialEq(bound = "")
```

Comment:
You can set the bound to `F: PrimeField, VerifierKey<F, PC>: foo`, where `foo \in {Debug, Eq, PartialEq}`.

Reply:
Hmm, I don't quite understand. The problem is that `PolynomialCommitment` and `PolynomialCommitment::Commitment` don't impl `Copy, Debug, Eq, PartialEq`. So no matter how we put the bounds in it'll never be satisfied.